pkg/pool/pool.go

```diff
@@ -0,0 +1,107 @@
+/*
+ Copyright 2023, NVIDIA CORPORATION & AFFILIATES
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package pool
+
+import (
+	"encoding/json"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	ipBlocksAnnotation = "ipam.nvidia.com/ip-blocks"
+)
```

Comment:
some duplication with Adrian's PR https://github.com/Mellanox/nvidia-k8s-ipam/blob/0a1f9841c3bd8790d7b9dc8511b059e7e50b5ec9/pkg/cni/pool/pool.go

Reply:
the PR is rebased, code duplication removed